src/main/java/io/dmitryivanov/crdt/helpers/Operations.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) pakoito 2016
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dmitryivanov.crdt.helpers;
+
+import java.util.*;
+
+public final class Operations {
+    private Operations() {
+        // No instances
+    }
+
+    public static <E> Set<E> diff(Set<E> firstSet, final Set<E> secondSet) {
+        return filtered(firstSet, new Predicate<E>() {
+            @Override
+            public boolean call(E element) {
+                return !secondSet.contains(element);
+            }
+        });
+    }
+
+    public static <E> Set<E> union(Set<E> firstSet, final Set<E> secondSet) {
+        final Set<E> newSet = new HashSet<>();
+        newSet.addAll(firstSet);
+        newSet.addAll(secondSet);
+        return newSet;
+    }
+
+    public static <E> Set<E> filtered(Set<E> set, Predicate<E> predicate) {
+        final Set<E> newSet = new HashSet<>();
+        for (E element : set) {
+            if (predicate.call(element)) {
+                newSet.add(element);
+            }
+        }
+        return newSet;
+    }
+
+    public static <E, R> Set<R> filteredAndMapped(Set<E> set, Predicate<E> predicate,
+            Mapper<E, R> mapper) {
+        final Set<R> newSet = new HashSet<>();
+        for (E element : set) {
+            if (predicate.call(element)) {
+                newSet.add(mapper.call(element));
+            }
+        }
+        return newSet;
+    }
+
+    public static <E> E select(Set<E> set, Predicate2<E, E> predicate) {
+        if (set.isEmpty()) {
+            throw new IllegalArgumentException("Empty set for select operation");
+        }
+        E winner = set.iterator().next();
+        for (E element : set) {
+            if (predicate.call(winner, element)) {
+                winner = element;
+            }
+        }
+        return winner;
+    }
+
+    public static <K, V> Map<K, Collection<V>> toListMap(Set<V> map, Mapper<V, K> mapper) {
```

Comment:
- isn't `groupBy` a better name for this kind of operation? 
- the first argument name is `Set<E> map` but can be just just a `Collection<E> collection`, I guess.